src/mono/mono/metadata/native-library.c

```diff
@@ -1219,9 +1219,11 @@ ves_icall_System_Runtime_InteropServices_NativeLibrary_FreeLib (gpointer lib, Mo
 		g_hash_table_add (native_library_module_blocklist, module);
 		mono_dl_close (module);
 	} else {
-		MonoDl raw_module = { { 0 } };
-		raw_module.handle = lib;
-		mono_dl_close (&raw_module);
+		MonoDl *raw_module = (MonoDl *) g_malloc0 (sizeof (MonoDl));
+		if (raw_module) {
+			raw_module->handle = lib;
+			mono_dl_close (raw_module);
+		}
```

Comment:
why?

Also, leaks the newly-allocated memory.

Reply:
mono_dl_close does a g_free as part of its implementation, that's what previously caused a corruption when using it as a stack variable, so the new allocated memory won't leak, it will be freed in mono_dl_close, https://github.com/dotnet/runtime/blob/4da6b9a8d55913c0ea560d63590d35dc942425be/src/mono/mono/utils/mono-dl.c#L391